src/test/java/bio/terra/service/snapshot/SnapshotExportIntegrationTest.java

```diff
@@ -234,19 +259,46 @@ public void snapshotExportTest() throws Exception {
             .getService();
 
     for (var path : paths) {
-      Blob blob = authedStorage.get(GcsUriUtils.parseBlobUri(parquet.getManifest()));
-      assertThat("Authorized user can read " + path, blob, notNullValue());
+      if (signUrls) {
+        // Make sure we can download the parquet bytes
+        byte[] bytes = readUrl(path);
+        assertThat(
+            "Signed URL is accessible and parquet isn't empty", bytes.length, greaterThan(0));
+      } else {
+        Blob blob = authedStorage.get(GcsUriUtils.parseBlobUri(parquet.getManifest()));
+        assertThat("Authorized user can read " + path, blob, notNullValue());
+
+        StorageException notAuthorizedException = null;
+        try {
+          unauthedStorage.get(GcsUriUtils.parseBlobUri(parquet.getManifest()));
+        } catch (StorageException ex) {
+          notAuthorizedException = ex;
+        }
+        // Appeasing SpotBugs
+        assert notAuthorizedException != null;
```

Comment:
Could `assertThrows` be used here? That should also make the linter happy, and avoid writing code that's partly copying what `assertThrows` does.

Also note that an assertion statement has no effect unless you run using `-enableassertions`. If you want it to do something, a function call is required.

Reply:
yeah good call